File: LeetCode/692/692.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        unordered_map<string, int> strings;
        for_each(words.begin(), words.end(),[&](const auto& elem){strings[elem]++;});
        auto cmp = [&](const auto& first, const auto& second){if(strings[first] == strings[second]) return first < second;
        else return strings[first] > strings[second];};
        vector<string> helper;
        for(const auto& [key, value] : strings) helper.emplace_back(key);
        sort(helper.begin(), helper.end(),cmp);
        vector<string> result;
        copy_n(helper.begin(), k, back_inserter(result));
        return result;
        
    }
};

File: LeetCode/692/692.rs
use std::collections::{BinaryHeap, HashMap};

impl Solution {
    pub fn top_k_frequent(words: Vec<String>, k: i32) -> Vec<String> 
    {
        let mut hash = HashMap::<String, i32>::new();
        words.into_iter().for_each(|x| *hash.entry(x).or_default() +=1);
        let mut vec: Vec<String> = hash.iter().map(|x| x.0).cloned().collect();
        vec.sort_by(|a,b| if hash.get(a).unwrap() == hash.get(b).unwrap()
        {
            return (*a).cmp(b);
        }
        else
        {
            return hash.get(b).unwrap().cmp(hash.get(a).unwrap());
        });
        vec.into_iter().take(k as usize).collect()
        
    }
}
